storage/range_test.go

```diff
@@ -160,6 +160,7 @@ func (tc *testContext) Start(t testing.TB) {
 			}
 		}
 		tc.store.ctx.DB = client.NewKV(nil, &testSender{store: tc.store})
+		tc.store.rangeGCQueue.db = tc.store.ctx.DB
```

Comment:
I think you can get the root of the problem more directly if you instead modify the testSender:

```go
// Create a dummy sender here. This makes sure that when the store is
// created, an actual object is passed down to the scanner queue (and 
// potentially others in the future); this is the best way of dealing with the
// circular dependency of creating a store using a sender which needs the
// store, the alternative being going through all dependent objects and
// adding in the DB client once it's in proper existence.
sender := &testSender{} // before line 146
tc.store.ctx.DB = client.NewKV(nil, sender)
...
// Now that we have our actual store, monkey patch the sender
// used in tc.store.DB.
sender.store = tc.store // after line 162
```

I might forget something, but if it works like this it's probably more future proof than figuring out where the DB gets passed down to and forgetting to add in new instances as they arise.